packages/server/src/api/students.rs

```diff
@@ -141,6 +141,19 @@ pub async fn compute_degree_status(mut user: User, db: Data<Db>) -> Result<HttpR
         .get_filtered::<Course>(catalog.get_all_course_ids(), FilterType::In, "_id")
         .await?;
 
+    // Fill tags for all student courses
+    user.details
+        .degree_status
+        .course_statuses
+        .iter_mut()
+        .for_each(|course_status| {
+            course_status.course.tags.as_mut().map(|_| {
```

Comment:
This is a bug!
Notice that `map` only takes the tags, ignores them (with `_`) and returns and Optional course **and then drops it immediately without mutating the tags of the actual course**!
I think what you meant is:
```rust
    user.details
        .degree_status
        .course_statuses
        .iter_mut()
        .for_each(|course_status| {
            course_status.course.tags = vec_courses
                    .iter()
                    .find(|course| course.id == course_status.course.id)
                    .map(|course| course.tags)
                    .flatten()
        });
```